src/Regridder.jl

```diff
@@ -0,0 +1,388 @@
+        # Note: out_np = degrees of freedom = polynomial degree + 1
+        kwargs = (; out_type = out_type, out_np = Nq)
+        kwargs = mono ? (; (kwargs)..., in_np = 1, mono = mono) : kwargs
+        remap_weights(
+            weightfile,
+            meshfile_rll,
+            meshfile_cgll,
+            meshfile_overlap;
+            kwargs...,
+        )
+        apply_remap(datafile_cgll, datafile_rll, weightfile, [varname])
+    else
+        @warn "Using the existing $datafile_cgll : check topology is consistent"
+    end
+
+    function get_time(ds)
+        if "time" in ds
+            data_dates =
+                Dates.DateTime.(
+                    reinterpret.(
+                        Ref(NCDatasets.DateTimeStandard),
+                        ds["time"][:],
+                    )
+                )
+        elseif "date" in ds
+            data_dates = strdate_to_datetime.(string.(ds["date"][:]))
+        else
+            @warn "No dates available in file $datafile_rll"
+            data_dates = [Dates.DateTime(0)]
+        end
+    end
+
+    # read the remapped file with sparse matrices
+    offline_outvector, times = NCDataset(datafile_cgll, "r") do ds_wt
+        (
+            offline_outvector = ds_wt[varname][:][:, :], # ncol, times
+            times = get_time(ds_wt),
+        )
+    end
+
+    # weightfile info needed to populate all nodes and save into fields with
+    #  sparse matrices
+    _, _, row_indices = NCDataset(weightfile, "r") do ds_wt
+        (Array(ds_wt["S"]), Array(ds_wt["col"]), Array(ds_wt["row"]))
+    end
+
+    target_unique_idxs =
+        out_type == "cgll" ?
+        collect(ClimaCore.Spaces.unique_nodes(space_undistributed)) :
+        collect(ClimaCore.Spaces.all_nodes(space_undistributed))
+    target_unique_idxs_i =
+        map(row -> target_unique_idxs[row][1][1], row_indices)
+    target_unique_idxs_j =
+        map(row -> target_unique_idxs[row][1][2], row_indices)
+    target_unique_idxs_e = map(row -> target_unique_idxs[row][2], row_indices)
+    target_unique_idxs =
+        (target_unique_idxs_i, target_unique_idxs_j, target_unique_idxs_e)
+
+    R = (; target_idxs = target_unique_idxs, row_indices = row_indices)
+
+    offline_field = ClimaCore.Fields.zeros(FT, space_undistributed)
+
+    offline_fields = ntuple(x -> similar(offline_field), length(times))
+
+    ntuple(
+        x -> reshape_cgll_sparse_to_field!(
+            offline_fields[x],
+            offline_outvector[:, x],
+            R,
+        ),
+        length(times),
+    )
+
+    # TODO: extend write! to handle time-dependent fields
+    map(
+        x -> write_to_hdf5(
+            REGRID_DIR,
+            hd_outfile_root,
+            times[x],
+            offline_fields[x],
+            varname,
+        ),
+        1:length(times),
+    )
+    jldsave(
+        joinpath(REGRID_DIR, hd_outfile_root * "_times.jld2");
+        times = times,
+    )
+end
+
+
+function hdread_regridfile(
```

Comment:
this function is a duplicate of `read_from_hdf5`. `read_from_hdf5` just has a name update to be consistent with `write_to_hdf5`, and adds a default value for `comms_ctx`. It's the more updated version that will be used by the coupler going forward.

Reply:
good catch!